src/legacy/core_plugins/kibana/public/management/sections/objects/saved_object_view.tsx

```diff
@@ -0,0 +1,176 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { Component } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiSpacer, EuiPageContent } from '@elastic/eui';
+import {
+  Capabilities,
+  SavedObjectsClientContract,
+  OverlayStart,
+  NotificationsStart,
+  SimpleSavedObject,
+} from '../../../../../../../core/public';
+import { ISavedObjectsManagementRegistry } from '../../saved_object_registry';
+import { Header, NotFoundErrors, Intro, Form } from './components/object_view';
+import { canViewInApp } from './lib/in_app_url';
+import { SubmittedFormData } from './types';
+
+interface SavedObjectEditionProps {
+  id: string;
+  serviceName: string;
+  serviceRegistry: ISavedObjectsManagementRegistry;
+  capabilities: Capabilities;
+  overlays: OverlayStart;
+  notifications: NotificationsStart;
+  notFoundType?: string;
+  savedObjectsClient: SavedObjectsClientContract;
+}
+
+interface SavedObjectEditionState {
+  type: string;
+  object?: SimpleSavedObject<any>;
+}
+
+export class SavedObjectEdition extends Component<
+  SavedObjectEditionProps,
+  SavedObjectEditionState
+> {
+  constructor(props: SavedObjectEditionProps) {
+    super(props);
+
+    const { serviceRegistry, serviceName } = props;
+    const type = serviceRegistry.get(serviceName)!.service.type;
+
+    this.state = {
+      object: undefined,
+      type,
+    };
+  }
+
+  componentDidMount() {
+    const { id, savedObjectsClient } = this.props;
+    const { type } = this.state;
+    savedObjectsClient.get(type, id).then(object => {
+      this.setState({
+        object,
+      });
+    });
+  }
+
+  render() {
+    const {
+      capabilities,
+      notFoundType,
+      serviceRegistry,
+      id,
+      serviceName,
+      savedObjectsClient,
+    } = this.props;
+    const { type } = this.state;
+    const { object } = this.state;
+    const { edit: canEdit, delete: canDelete } = capabilities.savedObjectsManagement as Record<
+      string,
+      boolean
+    >;
+    const canView = canViewInApp(capabilities, type);
+    const service = serviceRegistry.get(serviceName)!.service;
+
+    return (
+      <EuiPageContent horizontalPosition="center" data-test-subj="savedObjectsEdit">
+        <Header
+          canEdit={canEdit}
+          canDelete={canDelete}
+          canViewInApp={canView}
+          type={type}
+          onDeleteClick={() => this.delete()}
+          viewUrl={service.urlFor(id)}
+        />
+        {notFoundType && (
+          <>
+            <EuiSpacer size="s" />
+            <NotFoundErrors type={notFoundType} />
+          </>
+        )}
+        {canEdit && (
+          <>
+            <EuiSpacer size="s" />
+            <Intro />
+          </>
+        )}
+        {object && (
```

Comment:
Should we use Suspense or show a spinner while this loads?

Reply:
Will keep this one as a later improvement